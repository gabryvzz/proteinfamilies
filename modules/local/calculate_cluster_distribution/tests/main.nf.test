nextflow_process {

    name "Test Process CALCULATE_CLUSTER_DISTRIBUTION"
    script "../main.nf"
    process "CALCULATE_CLUSTER_DISTRIBUTION"

    test("clustering - tsv") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'clustering' ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("clustering - tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'clustering' ], // meta map
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
