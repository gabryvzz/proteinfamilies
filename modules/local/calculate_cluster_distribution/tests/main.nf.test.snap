{
    "clustering - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "clustering"
                        },
                        "clustering_clustering_distribution_mqc.csv:md5,dc4ff1f6c598621bbb48b667646e3665"
                    ]
                ],
                "1": [
                    "versions.yml:md5,88b6cac847ae3f6da84dbe86c343b4fe"
                ],
                "mqc": [
                    [
                        {
                            "id": "clustering"
                        },
                        "clustering_clustering_distribution_mqc.csv:md5,dc4ff1f6c598621bbb48b667646e3665"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88b6cac847ae3f6da84dbe86c343b4fe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T12:34:03.02050904"
    },
    "clustering - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "clustering"
                        },
                        "clustering_clustering_distribution_mqc.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,88b6cac847ae3f6da84dbe86c343b4fe"
                ],
                "mqc": [
                    [
                        {
                            "id": "clustering"
                        },
                        "clustering_clustering_distribution_mqc.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,88b6cac847ae3f6da84dbe86c343b4fe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T12:34:07.675502895"
    }
}